app/misc/git.ts

```diff
@@ -1317,29 +1317,29 @@ function requestLinkModal() {
 }
 
 /**
- * This method is called when a valid URL is given via the fetch-modal, and runs the
- * series of git commands which fetch and merge from an upstream repository.
+ * This method is called when the user clicks the "Remote" button on the navbar.
+ */
+function addRemoteModal() {
+  $('#add-remote-modal').modal('show');
+}
+
+/**
+ This function is called after the user enters the address of the upstream repo to sync from. It sets the
+ entered address as the upstream repo of the forked repository.
  */
-function fetchFromOrigin() {
-  console.log("begin fetching");
-  let upstreamRepoPath = document.getElementById("origin-path").value;
-  if (upstreamRepoPath != null) {
+function addRemoteRepo() {
+  let repository;
+  let upstreamRepoPath = document.getElementById("remote-path").value;
+  addCommand("git remote add upstream " + upstreamRepoPath);
+  if(upstreamRepoPath != null) {
     Git.Repository.open(repoFullPath)
       .then(function (repo) {
-        console.log("fetch path valid")
-        displayModal("Beginning Synchronisation...");
-        addCommand("git remote add upstream " + upstreamRepoPath);
-        addCommand("git fetch upstream");
-        addCommand("git merge upstrean/master");
-        console.log("fetch successful")
-        updateModalText("Synchronisation Successful");
-        refreshAll(repo);
-      },
-        function (err) {
-          console.log("Waiting for repo to be initialised");
-          displayModal("Please select a valid repository");
-        });
-  } else {
-    displayModal("No Path Found.")
+      repository = repo;
+      var result = Git.Remote.createWithFetchspec(repository, "upstream", upstreamRepoPath, 'remotes/upstream/master');
```

Comment:
Do we plan on extending this dialog in the future? 'git remote' has a lot more than setting upstream/master.

If no future plans, then perhaps rename this to button to Set Upstream Master? Or Set Upstream?

Reply:
No this is only going to set the upstream. I can rename it

Reply:
As for being able to view/delete the upstream we think we can implement these in the next sprint.

Reply:
@roweba please submit a new issue to track these new features.

Reply:
Issue #132 and issue #133 have been created.